server/interceptor/interceptor.go

```diff
@@ -140,6 +159,21 @@ func parseFullMethod(fullMethod string) (string, string, error) {
 	return tmp[0], tmp[1], nil
 }
 
+func userAndGlobal(kind quota.Kind, user string) []quota.Spec {
+	return []quota.Spec{
+		{Group: quota.User, Kind: kind, User: user},
+		{Group: quota.Global, Kind: kind},
+	}
+}
+
+func userTreeAndGlobal(kind quota.Kind, user string, treeID int64) []quota.Spec {
```

Comment:
These functions aren't helping readability at the moment; maybe either
 - add `..Spec` or `..Quota` or somesuch to their names, or
 - just fold the code inline above (my preference)